dashboard/app/assets/javascripts/client_state.js

```diff
@@ -42,15 +42,16 @@ dashboard.clientState.reset = function() {
 };
 
 /**
- * Returns the client-cached copy of the level source for the given
- * scriptLevelId, if it's newer than the given timestamp.
- * @param {number} scriptLevelId
+ * Returns the client-cached copy of the level source for the given script
+ * level, if it's newer than the given timestamp.
+ * @param {string} scriptName
+ * @param {string} levelKey
  * @param {number=} timestamp
  * @returns {string|undefined} Cached copy of the level source, or undefined if
  *   the cached copy is missing/stale.
  */
-dashboard.clientState.sourceForLevel = function (scriptLevelId, timestamp) {
-  var data = localStorage.getItem('source' + scriptLevelId);
+dashboard.clientState.sourceForLevel = function (scriptName, levelKey, timestamp) {
+  var data = localStorage.getItem('source_' + scriptName + '_' + levelKey);
```

Comment:
If source_scriptname_level is getting used all over the place then make a private function to standardize this naming